rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -0,0 +1,1424 @@
+            self.component_config
+        )
+        self.component_config = train_utils.update_evaluation_parameters(
+            self.component_config
+        )
+
+    # package safety checks
+    @classmethod
+    def required_packages(cls) -> List[Text]:
+        return ["tensorflow"]
+
+    def __init__(
+        self,
+        component_config: Optional[Dict[Text, Any]] = None,
+        inverted_label_dict: Optional[Dict[int, Text]] = None,
+        inverted_tag_dict: Optional[Dict[int, Text]] = None,
+        model: Optional[RasaModel] = None,
+        batch_tuple_sizes: Optional[Dict] = None,
+    ) -> None:
+        """Declare instance variables with default values."""
+
+        super().__init__(component_config)
+
+        self._check_config_parameters()
+
+        # transform numbers to labels
+        self.inverted_label_dict = inverted_label_dict
+        self.inverted_tag_dict = inverted_tag_dict
+
+        self.model = model
+
+        # encode all label_ids with numbers
+        self._label_data = None  # RasaModelData
+
+        # keep the input tuple sizes in self.batch_in
+        self.batch_tuple_sizes = batch_tuple_sizes
+
+        # number of entity tags
+        self.num_tags = 0
+
+        self.data_example = None  # Dict[Text, List[np.ndarray]]
+
+    @property
+    def label_key(self) -> Optional[Text]:
+        return "label_ids" if self.component_config[INTENT_CLASSIFICATION] else None
+
+    @staticmethod
+    def model_class() -> Any:
+        return DIET
+
+    # training data helpers:
+    @staticmethod
+    def _create_label_id_dict(
+        training_data: TrainingData, attribute: Text
+    ) -> Dict[Text, int]:
+        """Create label_id dictionary"""
+
+        distinct_label_ids = {
+            example.get(attribute) for example in training_data.intent_examples
+        } - {None}
+        return {
+            label_id: idx for idx, label_id in enumerate(sorted(distinct_label_ids))
+        }
+
+    def _create_tag_id_dict(self, training_data: TrainingData) -> Dict[Text, int]:
+        """Create label_id dictionary"""
+
+        if self.component_config[BILOU_FLAG]:
+            return bilou_utils.build_tag_id_dict(training_data)
+
+        distinct_tag_ids = set(
+            [
+                e["entity"]
+                for example in training_data.entity_examples
+                for e in example.get(ENTITIES)
+            ]
+        ) - {None}
+
+        tag_id_dict = {
+            tag_id: idx for idx, tag_id in enumerate(sorted(distinct_tag_ids), 1)
+        }
+        tag_id_dict["O"] = 0
+
+        return tag_id_dict
+
+    @staticmethod
+    def _find_example_for_label(
+        label: Text, examples: List[Message], attribute: Text
+    ) -> Optional[Message]:
+        for ex in examples:
+            if ex.get(attribute) == label:
+                return ex
+        return None
+
+    @staticmethod
+    def _find_example_for_tag(
```

Comment:
@tabergma we never use this method.
But in principle, I'd keep only `tag` here, as we do with `label`. Because as soon as we have them, they can be anything, not only entities

Reply:
if it is not used, we can remove it. we don't need it as we don't have features for tags, we simply use one-hot